src/client/remote_nodestream.ml

```diff
@@ -101,17 +108,18 @@ class remote_nodestream ((ic,oc) as conn) =
             loop_parts ()
           end
         | 3 ->
-          begin
-            Logger.debug_f_ "save_file" >>= fun () ->
-            Llio.input_string ic >>= fun name ->
-            Llio.input_int64 ic >>= fun length ->
-            Logger.info_f_ "got %s (%Li bytes)" name length >>= fun () ->
-            tlog_coll # save_tlog_file name length ic >>= fun () ->
-            loop_parts ()
-          end
+           begin
+             let () = _should_reinit := true in
+             Logger.debug_f_ "save_file" >>= fun () ->
+             Llio.input_string ic >>= fun name ->
+             Llio.input_int64 ic >>= fun length ->
+             Logger.info_f_ "got %s (%Li bytes)" name length >>= fun () ->
+             tlog_coll # save_tlog_file name length ic >>= fun () ->
```

Comment:
the name used to save this tlog file comes from the other side ... which may differ from the name that's needed here
there's no guarantee the names will be perfectly in sync
in theory it should even be possible to run the different nodes with different policies for when the rollover should occur?
(that would actually be a good test imo)